x-pack/plugins/lens/public/indexpattern_plugin/to_expression.ts

```diff
@@ -25,16 +23,20 @@ export function toExpression(state: IndexPatternPrivateState) {
     return operationDefinition.toEsAggsConfig(column, columnId);
   }
 
-  if (sortedColumns.length) {
-    const aggs = sortedColumns.map((col, index) => {
-      return getEsAggsConfig(col, state.columnOrder[index]);
+  const columnEntries = state.columnOrder.map(colId => [colId, state.columns[colId]]);
+
+  if (columnEntries.length) {
+    const aggs = columnEntries.map(([colId, col]) => {
+      return getEsAggsConfig(col as IndexPatternColumn, colId as string);
     });
 
-    const idMap = state.columnOrder.reduce(
-      (currentIdMap, columnId, index) => ({
-        ...currentIdMap,
-        [`col-${index}-${columnId}`]: columnId,
-      }),
+    const idMap = columnEntries.reduce(
+      (currentIdMap, [colId], index) => {
+        return {
+          ...currentIdMap,
+          [`col-${index}-${colId}`]: colId as string,
+        };
+      },
```

Comment:
Do you want to correct the expression call bool parameters to 
```
	      metricsAtAllLevels=false
	      partialRows=false
```

here or should we do this is in a separate PR?